site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -20,19 +21,19 @@ public function run() {
                 $this->viewCourseMaterialsPage();
                 break;
             case 'delete_course_material_file':
-                $this->deleteCourseMaterialFile();
+                $this->deleteCourseMaterialFile($_REQUEST["path"]);
                 break;
             case 'delete_course_material_folder':
-                $this->deleteCourseMaterialFolder();
+                $this->deleteCourseMaterialFolder($_REQUEST["path"]);
                 break;
```

Comment:
while you're in this file, could you condense these two functions into one using an if statement to distinguish between the two cases?